src/app/parcels/batch/AddressEditCellInput.tsx

```diff
@@ -0,0 +1,209 @@
+import FreeFormTextInput from "@/components/DataInput/FreeFormTextInput";
+import { CenterComponent, GappedDiv } from "@/components/Form/formStyling";
+import { Button, Checkbox, FormControlLabel } from "@mui/material";
+import { useState } from "react";
+import { Address, BatchActionType, BatchTableDataState } from "@/app/parcels/batch/BatchTypes";
+import styled from "styled-components";
+import { ADDRESS_WIDTH } from "@/app/parcels/batch/ColumnWidths";
+import { postcodeRegex } from "@/app/clients/form/formSections/AddressCard";
+
+const GappedDivMargin = styled(GappedDiv)`
+    margin: 1rem;
+    width: calc(${ADDRESS_WIDTH}px - 2rem);
+    overflow: clip;
+`;
+
+const TopFreeFormTextInput = styled(FreeFormTextInput)`
+    margin-top: 0.5rem;
+`;
+
+interface AddressInputFields {
+    addressLine1?: string;
+    addressLine2?: string;
+    addressTown?: string;
+    addressCounty?: string;
+    addressPostcode?: string;
+}
```

Comment:
I think we can remove the need for this by just using `Partial<Address>`, which makes all of the properties optional.
`getInitialAddressFields` would then look like:
```
const getInitialAddressFields = (id: number, tableState: BatchTableDataState): Partial<Address> => {
    const currentRowAddress =
        id === 0
            ? tableState.overrideDataRow.data.client.address
            : tableState.batchDataRows[id - 1].data.client.address;
    return currentRowAddress ?? {};
};
```